### PR TITLE
NPM on package lock

### PR DESCRIPTION
**Pull request by DarthHater**
This allows the plugin to work with:

- npm shrinkwrap
- npm package-lock
- yarn lock

This covers the bases with NPM, allowing someone to have fun all across the board!

### CODE REVIEW

File: ext-src/packages/npm/NpmUtils.ts [hunk 1]
```diff
@@ -13,47 +13,166 @@
...
       }
-      //read npm-shrinkwrap.json
-      let obj = JSON.parse(fs.readFileSync(npmShrinkwrapFilename, "utf8"));
-      
-      return Promise.resolve(this.flattenAndUniqDependencies(obj));
     } catch (e) {
       return Promise.reject(
-        "npm shrinkwrap failed, try running it manually to see what went wrong:" +
-          e.message
+        `${manifestType} read failed, try running it manually to see what went wrong:` +
```
Comment (allenhsieh) [inline ext-src/packages/npm/NpmUtils.ts]: trivial: mix of concat and templating, would this work instead?`${manifestType} read failed, try running it manually to see what went wrong: ${e.stderr}`

File: ext-src/packages/npm/NpmUtils.ts [hunk 2]
```diff
@@ -13,47 +13,166 @@
...
+import * as path from 'path';
 import * as fs from "fs";
 import * as _ from "lodash";
 
 import { exec } from "../../utils/exec";
 import { PackageDependenciesHelper } from "../PackageDependenciesHelper";
 import { NpmPackage } from "./NpmPackage";
+import { NPM_SHRINKWRAP_JSON, YARN_LOCK, PACKAGE_LOCK_JSON } from "./NpmScanType";
+
+const NPM_SHRINKWRAP_COMMAND = `npm shrinkwrap`;
```
Comment (allenhsieh) [inline ext-src/packages/npm/NpmUtils.ts]: trivial: i think back ticks may be mostly for template literals, maybe single quotes for this and next two lines instead

File: ext-src/packages/npm/NpmUtils.ts (hunk 1 above)
Comment (DarthHater) [reply ext-src/packages/npm/NpmUtils.ts]: Very nice catch, @allenhsieh !

Comment (DarthHater) [reply ext-src/packages/npm/NpmUtils.ts]: In: 1c7f218

File: ext-src/packages/npm/NpmUtils.ts (hunk 2 above)
Comment (DarthHater) [reply ext-src/packages/npm/NpmUtils.ts]: In: 32cd19b

**Pull request merged**